ESP8266/ESP8266.cpp

```diff
@@ -215,7 +220,7 @@ void ESP8266::_packet_handler()
     uint32_t amount;
 
     // parse out the packet
-    if (!_parser.recv(",%d,%d:", &id, &amount)) {
+    if (!_parser.recv(",%d,%" SCNu32 ":" , &id, &amount)) {
```

Comment:
If it's size_t, then it wants to be "%zu", which newlib-nano also doesn't support. Only pedantic thing that would be supported is taking unsigned long and using "%lu".